addons/stock/models/product_strategy.py

```diff
@@ -59,15 +60,18 @@ def _domain_product_id(self):
     location_in_id = fields.Many2one(
         'stock.location', 'When product arrives in', check_company=True,
         domain="[('child_ids', '!=', False), '|', ('company_id', '=', False), ('company_id', '=', company_id)]",
-        default=_default_location_id, required=True, ondelete='cascade')
+        default=_default_location_id, required=True, ondelete='cascade', index=True)
     location_out_id = fields.Many2one(
         'stock.location', 'Store to', check_company=True,
-        domain="[('id', 'child_of', location_in_id), ('id', '!=', location_in_id), '|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```

Comment:
Why you removed this?

Reply:
Now you could set location_out same as location_in, you don't need necessarily find a child location.
But you remind me, I think we could use an onchange to make them the same by default. What do you think?

Reply:
What is the functional purpose of a putaway WH/Shelf1 -> WH/shelf1?

Reply:
Smart putaway can find a child location of WH/shelf1 (location_out).

Reply:
A child location okay but the exact same location is useless I think